DSharpPlus.Caching/Memory/MemoryCacheService.cs

```diff
@@ -0,0 +1,91 @@
+using System;
+using System.Threading.Tasks;
+
+using DSharpPlus.Caching.Abstractions;
+
+using Microsoft.Extensions.Caching.Memory;
+using Microsoft.Extensions.Options;
+
+namespace DSharpPlus.Caching.Memory
+{
+    /// <summary>
+    /// An implementation of <see cref="ICacheService"/> relying on a memory cache.
+    /// </summary>
+    public class MemoryCacheService : ICacheService
+    {
+        private readonly MemoryCacheOptions _options;
+        private readonly IMemoryCache _cache;
+
+        public MemoryCacheService(IOptions<MemoryCacheOptions> options, IMemoryCache cache)
+        {
+            _options = options.Value;
+            _cache = cache;
+        }
+
+        
+        /// <inheritdoc/>
+        public ValueTask CacheAsync<TItem>(object key, TItem value)
+        {
+            TimeSpan absolute = _options.AbsoluteExpirations.ContainsKey(typeof(TItem))
+                ? _options.AbsoluteExpirations[typeof(TItem)]
+                : _options.DefaultAbsoluteExpiration;
+
+            TimeSpan sliding = _options.SlidingExpirations.ContainsKey(typeof(TItem))
+                ? _options.SlidingExpirations[typeof(TItem)]
+                : _options.DefaultSlidingExpiration;
```

Comment:
Consider a helper method to clean this up.

Reply:
implemented